app/src/main/java/com/example/calculaterserg/MainActivity.java

```diff
@@ -3,12 +3,175 @@
+    private Button button_plus;
+    private Button button_minus;
+    private Button button_multi;
+    private Button button_share;
+    private Button button_erase;
+    private TextView mNumberTextView;
+    private static String stringNum = "0";
+    private static int strToInt;
+    private static int firstNum;
+    private static String lastOperPressed = "None";
+
+    private static final String TAG = "MainActivity";
+    private static final String KEY_INDEX = "index";
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putInt(KEY_INDEX, strToInt); }
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_start);
+        setContentView(R.layout.activity_start_cons);
+        if (savedInstanceState != null) {
+            strToInt = savedInstanceState.getInt(KEY_INDEX);
+            stringNum = Integer.toString(strToInt);
+        }
+        mNumberTextView = (TextView) findViewById(R.id.memory_screen);
+        mNumberTextView.setText(stringNum);
+        button_9 = (Button) findViewById(R.id.button_9);
+        numUpdate(button_9, "9");
+        button_8 = (Button) findViewById(R.id.button_8);
+        numUpdate(button_8, "8");
+        button_7 = (Button) findViewById(R.id.button_7);
+        numUpdate(button_7, "7");
+        button_6 = (Button) findViewById(R.id.button_6);
+        numUpdate(button_6, "6");
+        button_5 = (Button) findViewById(R.id.button_5);
+        numUpdate(button_5, "5");
+        button_4 = (Button) findViewById(R.id.button_4);
+        numUpdate(button_4, "4");
+        button_3 = (Button) findViewById(R.id.button_3);
+        numUpdate(button_3, "3");
+        button_2 = (Button) findViewById(R.id.button_2);
+        numUpdate(button_2, "2");
+        button_1 = (Button) findViewById(R.id.button_1);
+        numUpdate(button_1, "1");
+        button_0 = (Button) findViewById(R.id.button_0);
+        numUpdate(button_0, "0");
+        button_plus = (Button) findViewById(R.id.button_plus);
+        operUpdate(button_plus, "addition");
+        button_minus = (Button) findViewById(R.id.button_minus);
+        operUpdate(button_minus, "subtraction");
+        button_multi = (Button) findViewById(R.id.button_multi);
+        operUpdate(button_multi, "multiplication");
+        button_share = (Button) findViewById(R.id.button_share);
+        operUpdate(button_share, "division");
+        button_equals = (Button) findViewById(R.id.button_equals);
+        equalsUpdate(button_equals);
+    }
+    public void numUpdate(final Button buttonName, final String num){
+        buttonName.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                if(lastOperPressed == "add" || lastOperPressed == "subtract" || lastOperPressed == "multiply" || lastOperPressed == "divide"){
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.